design-documents/graph-ql/coverage/returns.graphqls

```diff
@@ -0,0 +1,210 @@
+type Query{
+    # See https://github.com/magento/architecture/blob/master/design-documents/graph-ql/framework/attributes-metadata.md
+    pageSpecificCustomAttributes(
+        page_type: CustomAttributesPageEnum!
+    ): CustomAttributeMetadata
+}
+
+type Mutation {
+    requestReturn(input: RequestReturnInput!): RequestReturnOutput @doc(description: "Create a new return.")
+    addReturnComment(input: AddReturnCommentInput!): AddReturnCommentOutput @doc(description: "Add a comment to an existing return.")
+    addReturnTracking(input: AddReturnTrackingInput!): AddReturnTrackingOutput @doc(description: "Add tracking information to the return.")
+    removeReturnTracking(input: RemoveReturnTrackingInput!): RemoveReturnTrackingOutput @doc(description: "Remove a single tracked shipment from the return.")
+}
+
+input RequestReturnInput {
+    items: [RequestReturnItemInput!]!
+    comment_text: String
+}
+
+input RequestReturnItemInput {
+    order_item_id: ID! @doc(description: "ID of the order associated with the return.")
+    quantity_to_return: Float! @doc(description: "The quantity of the item requested to be returned.")
+    selected_custom_attributes: [ID!] @doc(description: "Values of return item attributes defined by the merchant, e.g. select attributes.")
+    entered_custom_attributes: [EnteredCustomAttributeInput!] @doc(description: "Values of return item attributes defined by the merchant, e.g. file or text attributes.")
+}
+
+input EnteredCustomAttributeInput {
+    id: ID!
+    value: String!
+}
+
+type RequestReturnOutput {
+    return: Return
+    returns(
+        pageSize: Int = 20 @doc(description: "Specifies the maximum number of results to return at once. Defaults to 20."),
+        currentPage: Int = 1 @doc(description: "Specifies which page of results to return. The default value is 1."),
+    ): Returns @doc(description: "Information about the customer returns.")
+}
+
+input AddReturnCommentInput {
+    return_id: ID!
+    comment_text: String!
+}
+
+type AddReturnCommentOutput {
+    return: Return
+}
+
+input AddReturnTrackingInput {
+    return_id: ID!
+    carrier_id: ID!
```

Comment:
This is an ID returned as part of `ReturnShippingCarrier` while using `available_shipping_carriers` on `Return` type. Should be used for return tracking input forms. and may be used by client to cache available carriers.

Reply:
Yes, I get it but how it will be represented? Usually, we use a carrier code + shipping method code to represent the delivery method. We do not store all carriers in the storage.

Reply:
It is up to the implementor, from the API point of view it does not matter because ID field is not expected to be human-readable.

Reply:
It's not up to the implementor, we have logic in Sales module which depends on the format of how shipping method and carrier are stored and represented, currently, it's `carrier_method`. For RMA functionality it should be consistent.

Reply:
From what I'm seeing, I think y'all might be discussing slightly separate things.

I think @paliarush is trying to make it clear that the client doesn't need to know how this ID was generated, because it's just an opaque reference for that object in the graph.

I think @joni-jones is trying to say that the implementor of the resolver in Magento core will need to make sure they're encoding/decoding the `ID` value in a similar manner as it's done in other places in the schema.

How wrong am I? 😄

Reply:
@DrewML you are right) I don't disagree with Alex. My point is following, as our GQL schema not only a schema for storefront developers but also an implementation guide for backend developers, some things we need to say explicitly. Like we did for order/invoice/shipment entities ID field.

Reply:
Definitely! Think we've been trying to get better about using inline comments (`#`) in SDL files for implementation-specific notes, and the `description` for consumer-facing docs.